cio/io_windows.go

```diff
@@ -43,22 +42,22 @@ func NewFIFOSetInDir(_, id string, terminal bool) (*FIFOSet, error) {
 	}, nil), nil
 }
 
-func copyIO(fifos *FIFOSet, ioset *Streams) (*cio, error) {
-	var (
-		set []io.Closer
-	)
+func copyIO(fifos *FIFOSet, ioset *Streams) (cios *cio, retErr error) {
+	cios = &cio{config: fifos.Config}
+
+	defer func() {
+		if retErr != nil {
+			_ = cios.Close()
+			cios = nil
+		}
+	}()
 
 	if fifos.Stdin != "" {
 		l, err := winio.ListenPipe(fifos.Stdin, nil)
 		if err != nil {
-			return nil, errors.Wrapf(err, "failed to create stdin pipe %s", fifos.Stdin)
+			return cios, errors.Wrapf(err, "failed to create stdin pipe %s", fifos.Stdin)
```

Comment:
Would it be simpler to not use a named return value for cio (`_ *cio, retErr error`), then just return `nil` here instead? Seems like we would prefer to return `nil` when we have an error, and I think we're only returning this due to the named return value.

Reply:
That's roughly what's in the 3rd commit, but does mean that we have to do the closing by iterating over the `set` slice; I thought we might as well use `cio.Close()` to do that for us (and skip the intermediate `set` slice)

Reply:
Oh! I think I understand what you mean; so make `cios` a local variable (but still use it in the `defer`); is that what you meant?

Reply:
Yeah, something like this:

```
func copyIO(fifos *FIFOSet, ioset *Streams) (_ *cio, retErr error) { // cios changed to _
	cios := &cio{config: fifos.Config} // = changed to :=

[...]

			return nil, errors.Wrapf(err, "failed to create stdin pipe %s", fifos.Stdin)
```

Reply:
That also lets you simplify the defer to just:

```
	defer func() {
		if retErr != nil {
			_ = cios.Close()
		}
	}()
```

Reply:
Ah, yes, I like that.

technically, we shouldn't have to set it to `nil` as no consumer should handle the value on error, but I wanted to keep the behavior as it was, just in case.

I'll have a look at updating the PR; thanks!